add(capability): monitoring capabilities and metadata

rkenefeck opened the pull request:
Add metadata and capabilities for Monitoring services

smendis-scottlogic commented on services/management/monitoring/metadata.yaml:
```diff
@@ -0,0 +1,11 @@
+  description: |
+    Provides the ability to continuously observe, track, and analyze
+      the performance, availability, and health of the service resources or
+      applications.
```
Minor: can you please fix the indentation here?

smendis-scottlogic commented:
Other than the above features can you check whether these capabilities also can be included under Monitoring please?

- Log Querying - Search & analyze logs with a query language.
- Alerts - Define rules to notify when a condition is matched
- Triggering - Trigger actions based on events.
- Custom Metrics - Ability to ingest app-specific or business metrics.
- Anomaly Detection - ML-based automatic detection of unusual metric patterns.
- Integration with Third-Party Tools - Connectors for Slack, PagerDuty, Jira, etc.

rkenefeck commented:
thanks for the feedback @smendis-scottlogic I have some thoughts on these additional capabilities:

> * Log Querying - Search & analyze logs with a query language.
    Do you think Log Querying is a capability of Logging rather than Monitoring? Log querying can be used to generate custom metrics, and custom metrics then provide the link to monitoring, so Log-based Metrics could be considered a feature of a monitoring system?

> * Alerts - Define rules to notify when a condition is matched
 Alerting has been captured as a shared capability (CCC.F17)

> * Triggering - Trigger actions based on events.
Added in latest commit, 

> * Custom Metrics - Ability to ingest app-specific or business metrics.
  Do you feel this is substantially different to the description of Metric Collection? 

> * Anomaly Detection - ML-based automatic detection of unusual metric patterns.
I debated this one myself! I think Anomaly Detection is common across all the observability family, and was still considering if it was wider than that and covered more service-families as well. Do you have any thoughts on whether this would become a common capability rather than a specific monitoring capability?

> * Integration with Third-Party Tools - Connectors for Slack, PagerDuty, Jira, etc.
Added in latest commit.

rkenefeck commented on services/management/monitoring/capabilities.yaml:
```diff
@@ -0,0 +1,54 @@
+shared-capabilities:
+  - reference-id: CCC
+    identifiers:
```
Thanks, this makes it much more readable
